packages/fxa-auth-server/scripts/stripe-products-and-plans-to-firestore-documents/stripe-products-and-plans-converter.ts

```diff
@@ -0,0 +1,457 @@
+  private supportedLanguages: string[];
+  private log: Logger;
+  private stripeHelper: StripeHelper;
+  private paymentConfigManager: PaymentConfigManager;
+
+  constructor({
+    log,
+    stripeHelper,
+    supportedLanguages,
+  }: {
+    log: Logger;
+    stripeHelper: StripeHelper;
+    supportedLanguages: string[];
+  }) {
+    this.log = log;
+    this.stripeHelper = stripeHelper;
+    this.supportedLanguages = supportedLanguages.map((l: string) =>
+      l.toLowerCase()
+    );
+    this.paymentConfigManager = Container.get(PaymentConfigManager);
+    this.paymentConfigManager.startListeners();
+  }
+
+  /**
+   * Get all of the metadata keys that start with the prefix, and
+   * concatenate them into a single list.
+   * */
+  getArrayOfStringsFromMetadataKeys(
+    metadata: Stripe.Product['metadata'],
+    metadataKeyPrefix: string
+  ): string[] {
+    const metadataValues = [];
+    for (const [metadataKey, metadataValue] of Object.entries(metadata)) {
+      if (metadataKey.startsWith(metadataKeyPrefix)) {
+        metadataValues.push(metadataValue);
+      }
+    }
+    return metadataValues;
+  }
+
+  /**
+   * Convert Stripe Product or Plan capabilities metadata keys into a
+   * nested CapabilityConfig object.
+   *
+   * For example, a Stripe object with the capabilities `metadata` keys:
+   *   {
+   *     capabilities: 'abc',
+   *     capabilities[clientId]: 'def',
+   *   }
+   * Would result in the following `capabilities` object:
+   *   {
+   *     '*': 'abc',
+   *     [clientId]: 'def',
+   *   }
+   */
+  capabilitiesMetadataToCapabilityConfig(
+    stripeObject: Stripe.Product | Stripe.Plan
+  ): CapabilityConfig {
+    const capabilities: CapabilityConfig = {};
+    for (const oldKey of Object.keys(stripeObject.metadata!).filter((key) =>
+      key.toLowerCase().startsWith('capabilities')
+    )) {
+      // Parse the key to determine if it's an 'all RP' or single RP capability
+      const [_, clientId] = oldKey.split(':');
+      const newKey = clientId ?? '*';
+      capabilities[newKey] = commaSeparatedListToArray(
+        stripeObject.metadata![oldKey]
+      );
+    }
+    return capabilities;
+  }
+
+  /**
+   * Convert Stripe Product or Plan style metadata keys into a
+   * StyleConfig object.
+   */
+  stylesMetadataToStyleConfig(
+    stripeObject: Stripe.Product | Stripe.Plan
+  ): StyleConfig {
+    const styleConfig: StyleConfig = {};
+    for (const key of StyleConfigKeys) {
+      const value = stripeObject.metadata![key];
+      if (value) {
+        styleConfig[key] = value;
+      }
+    }
+    return styleConfig;
+  }
+
+  /**
+   * Convert Stripe Product or Plan support metadata keys prefixed
+   * with 'support:' into a SupportConfig object.
+   */
+  supportMetadataToSupportConfig(
+    stripeObject: Stripe.Product | Stripe.Plan
+  ): SupportConfig {
+    const supportConfig: SupportConfig = {};
+    if (!stripeObject.metadata) {
+      return supportConfig;
+    }
```

Comment:
I noticed in some of the code above you asserted that the metadata property exists, but here you check and return early.  What's the difference?

Reply:
Huh; yeah I'm not being consistent. I can make sure I check for all of them. In some cases, I would pass the metadata on into another helper, and I would get a TS error as type `Plan.metadata` can be of type `Stripe.metadata` or `null`.